### ISSUE TITLE
Cannot create OrRequestMatcher with List.of(...)

### ISSUE THREAD

**Issue by sgraca** [spring-projects/spring-security#10703]
**Describe the bug**
The constructor for `OrRequestMatcher` fails with `NullPointerException` if the argument is a list created with Java's `List.of` method.

**To Reproduce**
```java
new OrRequestMatcher(List.of(new AntPathRequestMatcher("/test")));
```

**Expected behavior**
A working object is constructed

**Sample**

See steps to reproduce

Reports that include a sample will take priority over reports that do not.
At times, we may require a sample, so it is good to try and include a sample up front.

**Comment by sgraca** [comment]
@terminux Thanks for the explanation. That's exactly the same reason that I was able to find out. I used different list implementation to workaround the issue.

**Comment by jzheaux** [comment]
Thanks for the report, @sgraca. This is now addressed in aaaf7d35235bf19d8207e890830c1557dbc76a32 and has been backported.
